ui/pages/confirmations/confirmation/alerts/alerts-context.tsx

```diff
@@ -0,0 +1,89 @@
+import { ApprovalRequest } from '@metamask/approval-controller';
+import React, {
+  ReactElement,
+  createContext,
+  useCallback,
+  useContext,
+  useState,
+} from 'react';
+
+import useAlerts from '../../../../hooks/useAlerts';
+import { AlertActionHandlerProvider } from '../../../../components/app/alert-system/contexts/alertActionHandler';
+import { AlertMetricsProvider } from '../../../../components/app/alert-system/contexts/alertMetricsContext';
+import { ConfirmAlertModal } from '../../../../components/app/alert-system/confirm-alert-modal';
+import setAlerts from './setAlerts';
+
+type AlertContextType = {
+  hasAlerts: boolean;
+  showAlertsModal: () => void;
+};
+
+const NopeFunction = () => undefined;
+
+export const AlertContext = createContext<AlertContextType | undefined>(
```

Comment:
For clarity, is it worth renaming this and the file to something like `TemplateAlertContext` to differentiate it from the alert system itself? Maybe the provider also?

Reply:
fixed